### ISSUE TITLE
Two tasks that doesn't finish

### ISSUE THREAD

**Issue by ikergonzalez86** [elkarbackup/elkarbackup#128]
Good Afternoon

I have a problem with Elkarbackup

I have two tasks that doesn't finish (Always in state RUNNING). It seems that all files are copied but the state doesn't change. 

My Elkarbackup version is 1.1.5 on Debian 7.

What can i do?

Thanks

**Comment by xezpeleta** [comment]
Hi Iker,

Interesting, could it be the same bug reported in #82 ?

What happens when you try to run the job manually with "Run now" button? It's always in state RUNNING?

Please, add your `/var/log/elkarbackup/BnvLog.log` to check if we are talking about the same bug.

Thanks

**Comment by antton** [comment]
Hi @ikergonzalez86 

- How deep was your backup more or less? 
- Windows folders? 
- Linux directories? 
- How many Mb, Gb or Kb? 
- It was the first time or there where more previous backups?
- Your elkarbackup "backup" directory is on local server or NFS?

We need more information and also the log file that @xezpeleta was asking.

Thanks

**Comment by ikergonzalez86** [comment]
Hello

- One folder about 550GB and the other one 250 GB.
- The bigger is Windows Folders and the other one Linux Folders (moodledata)
- The backups are working from about two years until now.
- Backup directory is on local server.

Thanks

[BnvLog.txt](https://github.com/elkarbackup/elkarbackup/files/123591/BnvLog.txt)

**Comment by antton** [comment]
Sorry @ikergonzalez86 
Maybe It's an stupid question but what was first on the backup process? The Bigger Folder or the moodledata one?

Thanks!

**Comment by antton** [comment]
Hi @ikergonzalez86 

This is "out of topic"
I'm checking your log file and did realize about your number 10 client problem, It looks that you give an IP of an other old server, check your know /var/lib/elkarbackup/.ssh/known_hosts  on the 8 line.
"out of topic"

**Comment by xezpeleta** [comment]
Please @ikergonzalez86 , try to replace the file `/usr/share/elkarbackup/src/Binovo/ElkarBackupBundle/Lib/BackupRunningCommand.php` with the link below and try to backing it up again:

https://raw.githubusercontent.com/elkarbackup/elkarbackup/elkarbackup_1.1.3/src/Binovo/ElkarBackupBundle/Lib/BackupRunningCommand.php

By the way, I can see that job-32 (from client-13) needed 14 hours just for the **du** command...

That's why I think this bug is probably related with some change I made in version 1.1.4.


```sh
$ grep "du " BnvLog.txt
...
...
[2016-02-08 15:30:16] BnvWeb.INFO: Client "10", Job "28" du begin. {"source":"RunJobCommand","link":"/client/10/job/28"}  
[2016-02-08 15:30:46] BnvWeb.INFO: Client "10", Job "28" du end. {"source":"RunJobCommand","link":"/client/10/job/28"} 
 
[2016-02-08 15:32:20] BnvWeb.INFO: Client "10", Job "28" du begin. {"source":"RunJobCommand","link":"/client/10/job/28"}  
[2016-02-08 15:33:36] BnvWeb.INFO: Client "10", Job "28" du end. {"source":"RunJobCommand","link":"/client/10/job/28"} 
 
[2016-02-08 17:02:14] BnvWeb.INFO: Client "13", Job "33" du begin. {"source":"RunJobCommand","link":"/client/13/job/33"}  
[2016-02-08 17:14:32] BnvWeb.INFO: Client "13", Job "33" du end. {"source":"RunJobCommand","link":"/client/13/job/33"} 
 
[2016-02-08 18:00:11] BnvWeb.INFO: Client "14", Job "35" du begin. {"source":"TickCommand","link":"/client/14/job/35"}  
[2016-02-08 18:01:22] BnvWeb.INFO: Client "14", Job "35" du end. {"source":"TickCommand","link":"/client/14/job/35"}
  
[2016-02-08 18:05:12] BnvWeb.INFO: Client "11", Job "31" du begin. {"source":"TickCommand","link":"/client/11/job/31"}  
[2016-02-08 18:06:29] BnvWeb.INFO: Client "11", Job "31" du end. {"source":"TickCommand","link":"/client/11/job/31"}
  
[2016-02-08 18:10:09] BnvWeb.INFO: Client "1", Job "1" du begin. {"source":"TickCommand","link":"/client/1/job/1"}  
[2016-02-08 18:12:04] BnvWeb.INFO: Client "1", Job "1" du end. {"source":"TickCommand","link":"/client/1/job/1"} 
 
[2016-02-08 18:15:24] BnvWeb.INFO: Client "15", Job "37" du begin. {"source":"TickCommand","link":"/client/15/job/37"}  
[2016-02-08 18:15:36] BnvWeb.INFO: Client "15", Job "37" du end. {"source":"TickCommand","link":"/client/15/job/37"}
  
[2016-02-08 18:30:08] BnvWeb.INFO: Client "10", Job "28" du begin. {"source":"TickCommand","link":"/client/10/job/28"}  
[2016-02-08 18:30:45] BnvWeb.INFO: Client "10", Job "28" du end. {"source":"TickCommand","link":"/client/10/job/28"} 
 
[2016-02-08 18:31:35] BnvWeb.INFO: Client "10", Job "29" du begin. {"source":"TickCommand","link":"/client/10/job/29"}  
[2016-02-08 18:31:49] BnvWeb.INFO: Client "10", Job "29" du end. {"source":"TickCommand","link":"/client/10/job/29"} 
 
[2016-02-08 18:37:28] BnvWeb.INFO: Client "10", Job "30" du begin. {"source":"TickCommand","link":"/client/10/job/30"}  
[2016-02-08 18:43:22] BnvWeb.INFO: Client "10", Job "30" du end. {"source":"TickCommand","link":"/client/10/job/30"} 
 
[2016-02-08 20:00:16] BnvWeb.INFO: Client "6", Job "18" du begin. {"source":"TickCommand","link":"/client/6/job/18"}  
[2016-02-08 20:03:08] BnvWeb.INFO: Client "6", Job "18" du end. {"source":"TickCommand","link":"/client/6/job/18"}
  
[2016-02-08 20:03:18] BnvWeb.INFO: Client "6", Job "19" du begin. {"source":"TickCommand","link":"/client/6/job/19"}  
[2016-02-08 20:03:20] BnvWeb.INFO: Client "6", Job "19" du end. {"source":"TickCommand","link":"/client/6/job/19"}
  
[2016-02-08 20:06:38] BnvWeb.INFO: Client "6", Job "20" du begin. {"source":"TickCommand","link":"/client/6/job/20"}  
[2016-02-08 20:06:39] BnvWeb.INFO: Client "6", Job "20" du end. {"source":"TickCommand","link":"/client/6/job/20"} 
 
[2016-02-08 20:14:20] BnvWeb.INFO: Client "13", Job "32" du begin. {"source":"TickCommand","link":"/client/13/job/32"}  

[2016-02-08 21:00:09] BnvWeb.INFO: Client "8", Job "22" du begin. {"source":"TickCommand","link":"/client/8/job/22"}  
[2016-02-08 21:02:33] BnvWeb.INFO: Client "8", Job "22" du end. {"source":"TickCommand","link":"/client/8/job/22"}  

[2016-02-08 21:06:49] BnvWeb.INFO: Client "8", Job "23" du begin. {"source":"TickCommand","link":"/client/8/job/23"}  
[2016-02-08 21:06:53] BnvWeb.INFO: Client "8", Job "23" du end. {"source":"TickCommand","link":"/client/8/job/23"}  

[2016-02-08 21:58:24] BnvWeb.INFO: Client "3", Job "4" du begin. {"source":"TickCommand","link":"/client/3/job/4"}  
[2016-02-08 22:21:35] BnvWeb.INFO: Client "3", Job "4" du end. {"source":"TickCommand","link":"/client/3/job/4"}  

[2016-02-09 00:22:47] BnvWeb.INFO: Client "2", Job "2" du begin. {"source":"TickCommand","link":"/client/2/job/2"}  

[2016-02-09 06:14:40] BnvWeb.INFO: Client "7", Job "36" du begin. {"source":"TickCommand","link":"/client/7/job/36"}  
[2016-02-09 06:14:57] BnvWeb.INFO: Client "7", Job "36" du end. {"source":"TickCommand","link":"/client/7/job/36"}  

[2016-02-09 09:57:27] BnvWeb.INFO: Client "13", Job "32" du end. {"source":"TickCommand","link":"/client/13/job/32"}  

[2016-02-09 18:00:18] BnvWeb.INFO: Client "14", Job "35" du begin. {"source":"TickCommand","link":"/client/14/job/35"}  
[2016-02-09 18:02:19] BnvWeb.INFO: Client "14", Job "35" du end. {"source":"TickCommand","link":"/client/14/job/35"}  

[2016-02-09 18:05:14] BnvWeb.INFO: Client "11", Job "31" du begin. {"source":"TickCommand","link":"/client/11/job/31"}  
[2016-02-09 18:05:56] BnvWeb.INFO: Client "11", Job "31" du end. {"source":"TickCommand","link":"/client/11/job/31"}  

[2016-02-09 18:10:07] BnvWeb.INFO: Client "1", Job "1" du begin. {"source":"TickCommand","link":"/client/1/job/1"}  
[2016-02-09 18:12:04] BnvWeb.INFO: Client "1", Job "1" du end. {"source":"TickCommand","link":"/client/1/job/1"}  

[2016-02-09 18:16:21] BnvWeb.INFO: Client "15", Job "37" du begin. {"source":"TickCommand","link":"/client/15/job/37"}  
[2016-02-09 18:16:42] BnvWeb.INFO: Client "15", Job "37" du end. {"source":"TickCommand","link":"/client/15/job/37"}  

[2016-02-09 18:30:10] BnvWeb.INFO: Client "10", Job "28" du begin. {"source":"TickCommand","link":"/client/10/job/28"}  
[2016-02-09 18:31:48] BnvWeb.INFO: Client "10", Job "28" du end. {"source":"TickCommand","link":"/client/10/job/28"}  

[2016-02-09 18:33:27] BnvWeb.INFO: Client "10", Job "29" du begin. {"source":"TickCommand","link":"/client/10/job/29"}  
[2016-02-09 18:33:33] BnvWeb.INFO: Client "10", Job "29" du end. {"source":"TickCommand","link":"/client/10/job/29"}  

[2016-02-09 18:48:28] BnvWeb.INFO: Client "10", Job "30" du begin. {"source":"TickCommand","link":"/client/10/job/30"}  
[2016-02-09 19:00:10] BnvWeb.INFO: Client "10", Job "30" du end. {"source":"TickCommand","link":"/client/10/job/30"}
```

**Comment by antton** [comment]
Bloody hell!! 14 hours!

**Comment by ikergonzalez86** [comment]
First of all the moodledata folder and 3 hours later the other one.

The client 10 problem i solved it yesterday :)

Im going to try replacing the file. 

Thanks again

**Comment by xezpeleta** [comment]
Yes @antton , to be exact:

job-32: 14 hours
job-2: still waiting to finish du (after 19 hours)

I guess both (32 and 2) are the two problematic jobs.

**Comment by ikergonzalez86** [comment]
Good Afternoon

I try it and the task is OK but is still waiting because of du command.

[2016-02-10 14:41:03] BnvWeb.INFO: QUEUED {"source":"StatusReport","link":"/client/13/job/32"} root elkarbackup@cebanc.com
[2016-02-10 14:42:02] BnvWeb.INFO: QUEUED {"source":"StatusReport","link":"/client/13"}  
[2016-02-10 14:42:02] BnvWeb.INFO: QUEUED {"source":"StatusReport","link":"/client/13/job/32"}  
[2016-02-10 14:42:02] BnvWeb.INFO: RUNNING {"source":"StatusReport","link":"/client/13"}  
[2016-02-10 14:42:02] BnvWeb.INFO: RUNNING {"source":"StatusReport","link":"/client/13/job/32"}  
[2016-02-10 14:42:02] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.13_32.cfg" sync 2>&1 {"source":"RunJobCommand","link":"/client/13/job/32"}  
[2016-02-10 14:50:48] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.13_32.cfg" sync 2>&1 succeeded with output:  {"source":"RunJobCommand","link":"/client/13/job/32"}  
[2016-02-10 14:50:48] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.13_32.cfg" Daily 2>&1 {"source":"RunJobCommand","link":"/client/13/job/32"}  
[2016-02-10 15:04:38] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.13_32.cfg" Daily 2>&1 succeeded with output:  {"source":"RunJobCommand","link":"/client/13/job/32"}  
[2016-02-10 15:04:38] BnvWeb.INFO: Client "13", Job "32" ok. {"source":"RunJobCommand","link":"/client/13/job/32"}  
[2016-02-10 15:04:39] BnvWeb.INFO: OK {"source":"StatusReport","link":"/client/13/job/32"}  
[2016-02-10 15:04:39] BnvWeb.INFO: Client "13", Job "32" du begin. {"source":"RunJobCommand","link":"/client/13/job/32"}  

Thanks

**Comment by xezpeleta** [comment]
Then, as far as I can understand, the main problem here is the *du* command in large directories.

In version 1.1.3 or lower, *du* command was executed after job execution (once the status is changed to OK / FAIL).

By contrast, after version 1.1.4 *du* command is executed before post-scripts. That's why we see the job status as RUNNING if *du* is not finished.

If we don't find a better solution, maybe we should revert the changes made in 1.1.4.

**Comment by ikergonzalez86** [comment]
Thanks Xabi and Antton.

**Comment by ricardo777** [comment]
I also have issue with running the backup, I have 1 in queue and the rest is not running last date is from the first backup.

How does the backup run? in the log there are no errors.

Also have rebooted the server without change.

**Comment by xezpeleta** [comment]
> I have 1 in queue and the rest is not running last date is from the first backup.

@ricardo777 do you mean there is one job with "queued" status? or is it "running"?

**Comment by ricardo777** [comment]
@xezpeleta If I remember correctly it was queued

**Comment by xezpeleta** [comment]
@ricardo777 did you read all this issue? You should check the logfiles to understand what is happening.

If it's the same problem, you can try to solve it following my advice: https://github.com/elkarbackup/elkarbackup/issues/128#issuecomment-182020715

Anyway, hopefully this bug is already solved for the upcoming version (Elkarbackup v1.2.0). If you want to test it you can download it from the following link (but it's still a Beta release):

https://github.com/elkarbackup/elkarbackup/releases/tag/elkarbackup_1.2.0-beta1

**Comment by ricardo777** [comment]
@xezpeleta Yes, I think it has to do with the server I use it is a install from online.net and they block the root user as standard in the installation maybe it is because of this.

Maybe I will try a reinstall and if I got problems will open a new issue.

**Comment by ricardo777** [comment]
@xezpeleta Small update with the beta release upgrade, the cron did run yesterday will monitor it but looks it is running.

**Comment by xezpeleta** [comment]
@ricardo777 what do you mean?

Is rsync process running? Or do you see the status "running" while the rsync process is finished?

Thanks

**Comment by xezpeleta** [comment]
This should be fixed on 1.2

**Comment by ricardo777** [comment]
@xezpeleta is does for me 1.2 is still running great
